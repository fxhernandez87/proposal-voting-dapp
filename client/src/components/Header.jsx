import React from 'react';
import PropTypes from "prop-types";
import User from '../assets/User.svg';

const Header = ({ userLoggedIn }) => (
  <div className="App-header">
    <div className="content">
      <img src={User} width={40} alt="user-logo" />
      <p>{userLoggedIn}</p>
    </div>
  </div>
);


Header.propTypes = {
  userLoggedIn: PropTypes.string,
};

Header.defaultProps = {
  userLoggedIn: null
};

export default Header;