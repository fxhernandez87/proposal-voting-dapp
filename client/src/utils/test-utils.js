import React from "react";
import { render } from 'react-testing-library'
import GlobalState from '../contexts/GlobalState';

const AllTheProviders = ({ children }) => {
  return (
    <GlobalState>
        {children}
    </GlobalState>
  )
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }