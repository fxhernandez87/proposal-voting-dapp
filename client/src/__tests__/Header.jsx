import React from 'react';
import { render, fireEvent, waitForElement, cleanup } from 'test-utils';
import 'jest-dom/extend-expect';
import Header from '../components/Header';

beforeEach(cleanup);

test('Should render a header with a title', async () => {
  // Arrange
  const userLoggedIn = 'hello';

  const {getByText, asFragment} = render(
    <Header
      userLoggedIn={userLoggedIn}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByText(/hello/i)).not.toBeNull();
});