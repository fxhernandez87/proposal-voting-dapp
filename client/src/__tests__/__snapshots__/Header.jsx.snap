// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a header with a title 1`] = `
<DocumentFragment>
  <div
    class="App-header"
  >
    <div
      class="content"
    >
      <img
        alt="user-logo"
        src="User.svg"
        width="40"
      />
      <p>
        hello
      </p>
    </div>
  </div>
</DocumentFragment>
`;
